llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -601,6 +607,10 @@ async def _step_worker(
                     service_definition.name, service_definition.default_value
                 )
                 kwargs[service_definition.name] = service
+            for resource in config.resources:
+                kwargs[resource.name] = await resource.resource.call(
+                    name=resource.name, resource_manager=resource_manager
```

Comment:
This feels a bit convoluted. Can we make it something like
```py
kwargs[resource.name] = await resource_manager.get(resource.name)
```
And move the factory call into `get()`?